langchain4j-core/src/main/java/dev/langchain4j/data/message/AiMessage.java

```diff
@@ -61,6 +61,15 @@ public String text() {
         return text;
     }
 
+    /**
+     * Update the text of the message.
+     * Useful to update dynamic variables
+     */
+    // TODO: Discuss with langchain the best approach
+    public void updateText(String newText) {
```

Comment:
I am not sure this is a good idea

Reply:
I will try to find a better way, but i need to update the text of an AiMessage without knowing/changing the instance of the object AI message or my new AiStatMessage

Reply:
When i use parallel tools, I ask LLM to transform the request in list of tools calls, each tool produce a result that could be used in next tool. And at the end the final result could use previous result.
All this with just one call to LLM.
So I need to be able to pass the tokenUsage for the result to the final AiMessage (It's why I've created an AiStatsMessage) . And I need it to substitue the variable in the final result -> [here](https://github.com/langchain4j/langchain4j/pull/1353/files#diff-9c306fdf581c9261ea2907f36473d10b2b231f048fca19a7ca02784753a8562dR27)

Perhaps , I could just provide the method in AiMessage that throw an NotSupportException() by default. and Implement the update just for the Ollama AiStatsMessage ?